Framework.Core/System/Threading/Tasks/TaskCompletionSourceTheraotExtensions.cs

```diff
@@ -28,64 +29,41 @@ private static class TrySetCanceledCachedDelegate<T>
 
             private static Func<TaskCompletionSource<T>, CancellationToken, bool> CreateTrySetCanceledDelegate()
             {
-                var trySetCanceledCached = typeof(TaskCompletionSource<T>).GetMethod
-                (
+                var trySetCanceledMethod = typeof(TaskCompletionSource<T>).GetMethod(
```

Comment:
We do not need to do this every call.
Can we move it as static readonly field?

Reply:
there is only 1 if from ??= on each call
https://github.com/theraot/Theraot/blob/7f93fd84606bf195220b3a870f6fa0e41077ee34/Framework.Core/System/Threading/Tasks/TaskCompletionSourceTheraotExtensions.cs#L25-L30

Reply:
Ah . I see it is a lazy initialization.

Please mark unchanged variables as readonly so we don't change them.

Reply:
It's lazy inited, so it can't be readonly

Reply:
I see.